staging/src/k8s.io/api/resource/v1alpha3/types.go

```diff
@@ -0,0 +1,998 @@
+	// unknown identifiers then trigger a runtime error.
+	//
+	// The `cel.bind` function is enabled and can be used to simplify
+	// expressions that access multiple attributes with the same domain:
+	//
+	//    cel.bind(dra, device.attributes["dra.example.com"], dra.someBool && dra.anotherBool)
+	//
+	// Capacities associated with a device are made available through a
+	// nested `device.capacities` map the same way as attributes.
+	//
+	// The `device.driverName` string variable can be used to check for a specific
+	// driver explicitly in a filter that is meant to work for devices from
+	// different vendors. It is provided by Kubernetes and matches the
+	// `driverName` from the ResourceSlice which provides the device.
+	//
+	// The CEL expression is applied to *all* available devices from any driver.
+	// The expression has to check for existence of an attribute when it is not
+	// certain that it is provided because runtime errors are not automatically
+	// treated as "don't select device". Instead, device selection fails completely
+	// and reports the error.
+	//
+	// Some more examples:
+	//
+	//    "memory" in device.capacities["dra.example.com"] && # Is the capacity available?
+	//       device.capacities["dra.example.com"].memory.isGreaterThan(quantity("1Gi")) # >= 1Gi
+	//
+	//    device.attributes["dra.example.com"].driverVersion.isGreaterThan(semver("1.0.0")) # >= v1.0.0, runtime error if not available
+	//
+	//    device.driverName == "dra.example.com" # any device from that driver
+	//
+	// +required
+	Expression string `json:"expression" protobuf:"bytes,1,name=expression"`
+}
+
+// Besides the request name slice, constraint must have exactly one field set.
+type Constraint struct {
+	// The constraint applies to devices in these requests. A single entry is okay
+	// and used when that request is for multiple devices.
+	//
+	// If empty, the constrain applies to all devices in the claim.
+	//
+	// +optional
+	// +listType=atomic
+	RequestNames []string `json:"requestNames,omitempty" protobuf:"bytes,1,opt,name=requestNames"`
+
+	// The devices must have this attribute and its value must be the same.
+	//
+	// For example, if you specified "dra.example.com/numa" (a hypothetical example!),
+	// then only devices in the same NUMA node will be chosen.
+	//
+	// +required
+	MatchAttribute *string `json:"matchAttribute,omitempty" protobuf:"bytes,2,opt,name=matchAttribute"`
+
+	// TODO (?)
+	// MatchQuantity *string
+
+	// Future extension, not part of the current design:
+	// A CEL expression which compares different devices and returns
+	// true if they match.
+	//
+	// Because it would be part of a one-of, old schedulers will not
+	// accidentally ignore this additional, for them unknown match
+	// criteria.
+	//
+	// matcher string
+}
+
+// ClaimConfiguration is used for configuration parameters in ResourcClaimSpec.
+type ClaimConfiguration struct {
+	// The configuration applies to devices in these requests.
```

Comment:
Again, just devices?  or devices and Gizmos?

Reply:
I left out the "devices" part entirely:
> Requests lists the names of requests where the configuration applies.

This is consistent with defining a mechanism that can be used for both.